src/MongoDB.Driver/MongoClientSettings.cs

```diff
@@ -610,7 +615,7 @@ public bool VerifySslCertificate
             set
             {
                 if (_isFrozen) { throw new InvalidOperationException("MongoClientSettings is frozen."); }
-                _allowInsecureTls = !value;
+                AllowInsecureTls = !value;
```

Comment:
The fact that you had to use the property instead of the backing field is a bad sign.

At the very least add a comment:

    AllowInsecureTls = !value; // use property instead of private field because setter has additional side effects

The fact that a comment is required is also a bad sign.